log4j-api/src/main/java/org/apache/logging/log4j/status/StatusLogger.java

```diff
@@ -16,311 +16,590 @@
+            this.bufferCapacity = readBufferCapacity(fileProvidedProperties);
+            this.defaultListenerLevel = readDefaultListenerLevel(fileProvidedProperties);
+            this.instantFormatter = readInstantFormatter(fileProvidedProperties);
+        }
+
+        /**
+         * Gets the static instance.
+         *
+         * @return a singleton instance
+         */
+        public static Config getInstance() {
+            return INSTANCE;
+        }
+
+        private static boolean readDebugEnabled(final Properties fileProvidedProperties) {
+            final String debug = readProperty(fileProvidedProperties, DEBUG_PROPERTY_NAME);
+            return debug != null;
+        }
+
+        private static int readBufferCapacity(final Properties fileProvidedProperties) {
+            final String capacityString = readProperty(fileProvidedProperties, BUFFER_CAPACITY_PROPERTY_NAME);
+            return capacityString != null ? Integer.parseInt(capacityString) : BUFFER_CAPACITY_DEFAULT_VALUE;
+        }
 
-    private static final String NOT_AVAIL = "?";
+        private static Level readDefaultListenerLevel(final Properties fileProvidedProperties) {
+            final String level = readProperty(fileProvidedProperties, DEFAULT_LISTENER_LEVEL_PROPERTY_NAME);
+            return level != null ? Level.valueOf(level) : DEFAULT_LISTENER_LEVEL_DEFAULT_VALUE;
+        }
 
-    static final PropertiesUtil PROPS = new PropertiesUtil("log4j2.StatusLogger.properties");
+        private static DateTimeFormatter readInstantFormatter(final Properties fileProvidedProperties) {
+            final String format = readProperty(fileProvidedProperties, INSTANT_FORMAT_PROPERTY_NAME);
+            return format != null ? DateTimeFormatter.ofPattern(format) : null;
+        }
 
-    private static final int MAX_ENTRIES = PROPS.getIntegerProperty(MAX_STATUS_ENTRIES, 200);
+        private static String readProperty(final Properties fileProvidedProperties, final String propertyName) {
+            final String systemProvidedValue = System.getProperty(propertyName);
+            return systemProvidedValue != null
+                    ? systemProvidedValue
+                    : (String) fileProvidedProperties.get(propertyName);
+        }
 
-    private static final String DEFAULT_STATUS_LEVEL = PROPS.getStringProperty(DEFAULT_STATUS_LISTENER_LEVEL);
+        // We need to roll out our own `.properties` reader.
+        // We could have used `PropertiesUtil`, `PropertyFilePropertySource`, etc.
+        // Consequently, they would delegate to `LoaderUtil`, etc.
+        // All these mechanisms expect a working `StatusLogger`.
+        // Hence, in order to be self-sufficient, we cannot rely on them.
+        private static Properties readPropertiesFile() {
+            final Properties properties = new Properties();
+            final URL url = StatusLogger.class.getResource(PROPERTIES_FILE_NAME);
```

Comment:
This is an area I struggled with when I attempted a variation of this PR. The original `LoaderUtil` code attempts to find this file using multiple `ClassLoader` instances. At the very least, it could be useful to include a fallback here to try to load the file from the thread context class loader, though on the other hand, this is loaded so early in code that a custom TCCL is unlikely to have been set yet.

Reply:
The `log4j2.StatusLogger.properties` file provides options that apply to all logger contexts.
Looking it up in the TCCL does not make sense to me.

Reply:
I am taking the initiative to mark this conversation as resolved. @jvz, if you have objections, please feel free to re-open the conversation.

Reply:
I believe this line introduced a bug. Unlike [`ClassLoader.getResource`](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/ClassLoader.html#getResource(java.lang.String)), which takes absolute resource paths, [`Class.getResource`](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Class.html#getResource(java.lang.String)) supports relative resource paths, so the `log4j2.StatusLogger.properties` resource must be placed into jar resources as `org/apache/logging/log4j/status/log4j2.StatusLogger.properties`, and existing configuration resources which worked with previous log4j2 releases are ignored.
```suggestion
            final URL url = StatusLogger.class.getResource('/' + PROPERTIES_FILE_NAME);
```

Reply:
@carterkozak, doh! :facepalm: Thanks for the report. Could you review #2354, please?